src/ps/ps.cpp

```diff
@@ -972,11 +975,25 @@ PetscErrorCode PSSetUp(PS ps) {
   CHKERRQ(ierr);
 
   /* Set up edge connectivity */
-
+  /* Edges include true edges + dummy edges set up for isolated buses
+     These edges for isolated buses are inserted so that DMNetwork
+     correctly knows the number of vertices. DMNetwork filters out
+     any nodes that do not have edges and so this messes up our network
+     data layout that has these isolated buses. So, we create dummy
+     edges connecting these isolated buses. These are only used here
+     for the DMNetwork business, not anywhere else
+  */
   ierr = PetscCalloc1(2 * Nlines, &lineconn[0]);
   CHKERRQ(ierr);
   ierr = PSGetLineConnectivity(ps, Nlines, lineconn[0]);
   CHKERRQ(ierr);
+  /* Insert dummy edges for isolated_buses */
+  //  for(i=0; i < ps->nisolated_buses; i++) {
+  /* Each dummy line is connected between bus 0 and the isolated bus */
+  //    lineconn[0][2 * (Nlines+i)    ] = ps->isolated_buses[i];
+  //    lineconn[0][2 * (Nlines+i) + 1] = 0;
+  //  }
+  //  ierr = PetscFree(ps->isolated_buses);
```

Comment:
Are these meant to be commented out?

Reply:
I was working on a tricky case with isolated nodes that messes up the network layout. The comment above summarizes this. I will be working on this in a follow-up PR. So, let it be here. I plan to take care of it immediately after this PR.